redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/ProtocolState.kt

```diff
@@ -36,6 +38,48 @@ public class ProtocolState {
     changes.add(change)
   }
 
+  /**
+   * Each time we add a node, confirm it's preceded by either a [Create] or a [Remove].
+   *
+   * If we're adding a node that's been removed, the node should have been detached instead. Rewrite
+   * the change log to match what it should have been.
+   */
+  public fun requireCreateOrDetach(id: Id, widget: ProtocolWidget) {
```

Comment:
This fix is necessary to make it all work. It has the effect of replacing a remove with a detach